extensions/sql-database-projects/src/models/publishProfile/publishProfile.ts

```diff
@@ -129,7 +133,169 @@ async function readConnectionString(xmlDoc: any): Promise<{ connectionId: string
 /**
  * saves publish settings to the specified profile file
  */
-export async function savePublishProfile(profilePath: string, databaseName: string, connectionString: string, sqlCommandVariableValues?: Map<string, string>, deploymentOptions?: mssql.DeploymentOptions): Promise<void> {
+export async function savePublishProfile(profilePath: string, databaseName: string, connectionString: string, sqlCommandVariableValues?: Map<string, string>, deploymentOptions?: mssql.DeploymentOptions | vscodeMssql.DeploymentOptions): Promise<void> {
 	const dacFxService = await utils.getDacFxService();
-	await dacFxService.savePublishProfile(profilePath, databaseName, connectionString, sqlCommandVariableValues, deploymentOptions);
+	if (utils.getAzdataApi()) {
+		await (dacFxService as mssql.IDacFxService).savePublishProfile(profilePath, databaseName, connectionString, sqlCommandVariableValues, deploymentOptions as mssql.DeploymentOptions);
+	} else {
+		await (dacFxService as vscodeMssql.IDacFxService).savePublishProfile(profilePath, databaseName, connectionString, sqlCommandVariableValues, deploymentOptions as vscodeMssql.DeploymentOptions);
+	}
+}
+
+export function promptToSaveProfile(project: Project, publishProfileUri?: vscode.Uri) {
+	return vscode.window.showSaveDialog(
+		{
+			defaultUri: publishProfileUri ?? vscode.Uri.file(path.join(project.projectFolderPath, `${project.projectFileName}_1.publish.xml`)),
+			saveLabel: constants.save,
+			filters: {
+				'Publish Settings Files': ['publish.xml'],
+			}
+		}
+	);
+}
+
+/**
+ * Prompt to save publish profile and add to the tree
+ * @param project
+ * @param settings Publish settings
+ * @returns
+ */
+export async function promptForSavingProfile(project: Project, settings: ISqlProjectPublishSettings | ISqlDbDeployProfile | IPublishToDockerSettings | undefined) {
+	const result = await vscode.window.showInformationMessage(constants.saveProfile, constants.yesString, constants.noString);
+	if (result === constants.yesString) {
+		let publishProfileUri: vscode.Uri | undefined;
+		if (settings) {
+			if (ifIsISqlProjectPublishSettings(settings)) {
```

Comment:
can these helper functions be renamed to remove the "if" at the front? Looks a little strange having `if (if...)`